source/base/fileinputoutput.cpp

```diff
@@ -317,12 +319,19 @@ IStream *NewIStream(const Path& p, unsigned int stype)
         throw POV_EXCEPTION(kCannotOpenFileErr, str);
     }
 
-    return new IFileStream(p().c_str());
+    stream = new IFileStream(p().c_str());
+    if (!(*stream))
+    {
+        delete stream;
+        return NULL;
+    }
+    return stream;
```

Comment:
OK, it looks like my theory about the cast was wrong...
Prior to commit 051b674, returning NULL on failure actually was the behavior of NewIStream / NewOStream (and, as far as I can tell, had been since version 3.5).